devito/symbolics/manipulation.py

```diff
@@ -44,7 +44,7 @@ def _uxreplace(expr, rule):
     if expr in rule:
         v = rule[expr]
         if not isinstance(v, dict):
-            return v, True
+            return _uxreplace(v, rule)[0], True
```

Comment:
@FabioLuporini this makes the mapper processign nested so should handle any depth, will add a test for it.